File: PHP/narcissistic.php
<?php
/*
|--------------------------------------------------------------------------
| Does my number look big in this?
|--------------------------------------------------------------------------
|
*/

function narcissistic($value) {
   $str = str_split(strval($value));
   return array_sum(array_map(fn($i) => $i ** count($str), $str)) == $value;
}


function narcissistic(int $value): bool {
   $len = strlen($value);
   $sum = 0;
   foreach (str_split($value) as $num) {
      $sum += pow($num,$len);
   }
   return ($sum == $value);
}


function narcissistic(int $value): bool {
   $total = array_reduce(str_split($value), function ($carry, $item) use ($value) {
      $carry += pow($item, strlen($value));
      return $carry;
   }, 0);
   return $total === $value;
}


function narcissistic(int $value): bool {
   $power = (floor(log10($value) + 1));
   $sum = 0;
   $n = $value;
   while ($n > 0) {
      $digit = $n % 10;
      $sum += pow($digit, $power);
      $n = floor($n/10);
   }
   return $value == $sum ? true: false;
}


print_r(narcissistic(7). ' ' . true . '
');
print_r(narcissistic(371). ' ' . true . '
');

File: Python/6py 1/narcissistic.py
"""
|--------------------------------------------------------------------------
| Does my number look big in this?
|--------------------------------------------------------------------------
|
"""

def narcissistic( value ):
   string = str(value)
   res = 0
   for i in range(len(string)):
      res += int(string[i]) ** len(string)
   return res == value


def narcissistic(value):
   return value == sum(int(x) ** len(str(value)) for x in str(value))


def narcissistic( value ):
   digs = map(int, str(value))
   l = len(digs)
   return value == sum(map(lambda x: x**l, digs))


def narcissistic( value ):
   from math import log10
   c = int(log10(value)+1)
   def f(n): return (n % 10)**c + f(n/10) if n!=0 else 0
   return True if value==f(value) else False


def narcissistic( value ):
   s = str(value)
   return sum(map(lambda x: int(x) ** len(s), s)) == value 


print(narcissistic(7), True, '7 is narcissistic');
print(narcissistic(371), True, '371 is narcissistic');
print(narcissistic(122), False, '122 is not narcissistic')
print(narcissistic(4887), False, '4887 is not narcissistic')